### PR TITLE
Workbench apply settings without relaunch

### LINKED ISSUE #27861: Workbench Settings: Apply without restart
As a user I would like to change settings in workbench and have them applied to the running session if possible. This is possible with things like the invisible workspaces but not with settings such as the visible categories.

The "Save Settings" button is also a little ambiguous. Consider the more standard approach of separate apply and save buttons as this is what most users will be used to.

### LINKED ISSUE #27868: Workbench resets the default instrument to ALF in Settings
### Steps + behavior
Open Workbench
In File > Settings change the instrument to another (within ISIS)
Click Save settings (or whatever that button may now say!)
and close Settings and Mantid WB entirely
Reopen and notice that in Settings your defualt instrument is reset to ALF.
### Platforms affected
Workbench Py3 nightly

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**

The algorithms category and interface hiding should now apply as soon as the setting dialog is closed.
The button has also been changed to be close rather than "SaveSettings", as most of the settings were saved even if you clicked the X close button.

Also harmonised both the close and X close actions

Also resolved a bug where the setting dialog reset the default instrument to the first in the facility list on starting the settings dialog.

**To test:**
1. open workbench
1. File->Settings
1. Fiddle with categories, both algorithms and interface hiding
1. check that the workbench toolboxes refresh when settings is closed.
1. try with both close and X close
1. Show invisible workspaces should also be immediate.

Now check that the default instrument is preserved.
1. In settings set a default instrument that is NOT the first in the list
1. close settings
1. exit workbench
1. restart workbench
1. open settings, check that the default instrument is still set
1. close and reopen settings, make sure it has not sneakily changed on you
1. close and reopen settings, change it to another instrument
1. close and reopen settings, make sure it has not sneakily changed on you

Fixes #27861 and fixes #27868 
 

The updating is in the release notes, the default instrument bug isn't as I think we introduced it in this release

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/widgets/settings/presenter.py [hunk 1]
```diff
@@ -76,18 +75,18 @@ def action_section_changed(self, new_section_pos):
...
-
     def action_open_help_window(self):
         InterfaceManager().showHelpPage('qthelp://org.mantidproject/doc/workbench/settings.html')
 
-    def refresh_workspaces(self):
+    def view_closing(self):
         """
-        Refreshes the workspaces shown, so that if the invisible workspaces
-        setting was changed the effect will be reflected in the workspacewidget
+        Saes te mantid settings and updates updates te parent
```
Comment (Harrietbrown) [inline qt/applications/workbench/workbench/widgets/settings/presenter.py]: I think this is what it is supposed to mean.

```suggestion
        Saves the mantid settings and updates updates the parent
```

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
The settings take effect after closing the window. However closing the window is noticably slower even if no changes are made. If possible there should be a check to see if the setting have been changed before repopulating the algorithm widget and taskbar.

**Comment by NickDraper** [comment]
It's not really possible to check as most of the changes happen immediately, the dealy is from updating toolboxes in the workbench that might have been changed.

I'm not sure people fiddle in here often enough to be worried about a small change in closing speed

**Comment by NickDraper** [comment]
Reopened as I only closed it by mistake

**Review by Harrietbrown — APPROVED** [review]
In that case it is probably not worth making a check for that. The actual functionality is all good and the jenkins failure is unrelated.

**Pull request merged**